### ISSUE TITLE
Why does `nn` module need a higher learning rate?

### ISSUE THREAD

**Issue by Liwink** [jcjohnson/pytorch-examples#12]
Thanks a lot for your amazing examples!

I find that in the examples using `nn` module the `learning_rate` is 1e-4, while without the `nn` module the `learning_rate` is 1e-6.

I did not figure out the reason why those examples using `nn` require larger `learning_rate`🙁

**Comment by wenlibin02** [comment]
Part A. Using the code in the example. 
A.1 The result with learning_rate = 1E-6: 

	0   69.15245056152344
	50  68.89289855957031
	100 68.63490295410156
	150 68.3786392211914
	200 68.12423706054688
	250 67.87149047851562
	300 67.62022399902344
	350 67.37112426757812
	400 67.12361145019531
	450 66.8774642944336

A.2 The result with learning_rate = 1E-4:

	0   69.15245056152344
	50  49.68042755126953
	100 37.369651794433594
	150 28.201385498046875
	200 21.028329849243164
	250 15.453542709350586
	300 11.202285766601562
	350 8.05745792388916
	400 5.785703659057617
	450 4.168478012084961

Part B. Then I added a few lines to initialize the weights right after the definition of `model`. 

```python
def init_weights(m):
    if type(m) == torch.nn.Linear:
        m.weight.data = torch.randn(m.weight.data.shape)
model = model.apply(init_weights)
```

Now a 1E-6 learning rate gives, 

	0 2980668.25
	50 354273.59375
	100 114982.5859375
	150 49290.53515625
	200 24664.033203125
	250 13616.025390625
	300 8043.4990234375
	350 4996.1318359375
	400 3224.521728515625
	450 2142.218017578125

and a 1E-5 learning rate gives,

	0 2980668.25
	50 1551.1671142578125
	100 77.2059097290039
	150 6.307880401611328
	200 0.6416722536087036
	250 0.07231520116329193
	300 0.008584558963775635
	350 0.001067501143552363
	400 0.00015966703358571976
	450 3.668743738671765e-05

In summary, the differences between this model and the previous ones, 

- this model (the original code) does not initialize the weights
- the definition of the MSELoss may be different from previous (not quite sure)
- this model has bias

**Comment by csyanbin** [comment]
Hi, @wenlibin02.
I tested the sample code on Pytorch 1.0

(1) About weight initialization, you are right. The default initialization for nn.Linear is U(-sqrt(k), sqrt(k)), k = 1/n_input. The initial weight values are quite small compared with random_normal initialization. That's why lr=1e-6 cannot work for this example. 

(2) I monitored the following two losses.
`loss1 = loss_fn(y_pred, y)`
`loss2 = (y_pred-y).pow(2).sum()`
They give the same value for all cases.
